app/views/Project/Tagging/Sources/utils.tsx

```diff
@@ -46,3 +53,48 @@ export function transformSourcesFilterToEntriesFilter(filters: FaramValues) {
         leadAuthoringOrganizationTypes: authoringOrganizationTypes,
     };
 }
+
+export function getSortState(
+    orderingFilter: LeadOrderingEnum[] | null | undefined,
+): { name: string; direction: SortDirection } | undefined {
+    const [direction, ...orderingName] = orderingFilter?.[0] ? orderingFilter[0].split('_') : [];
+    if (direction) {
+        return ({
+            name: orderingName.join('_'),
+            direction: direction === 'ASC' ? SortDirection.asc : SortDirection.dsc,
+        });
+    }
+    return undefined;
+}
+
+function getDateString(dateTimeString: string | null | undefined) {
+    if (dateTimeString && Date.parse(dateTimeString)) {
+        return encodeDate(new Date(dateTimeString));
+    }
+    return dateTimeString;
+}
```

Comment:
Why are we not using a utils from fujs?

Reply:
we need to check if `dateTimeString` is valid date string value.

Reply:
Why do we have such cases?